v2: panic when using HTTP3 with self-signed certificates

Caddy version: `v2.0.0-beta.14`

Stack:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x17912e9]

goroutine 63 [running]:
github.com/mholt/certmagic.(*Config).getCertificate(0xc0007e8000, 0xc0007e20b0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/damnever/.go/pkg/mod/github.com/mholt/certmagic@v0.9.3/handshake.go:101 +0x2d9
github.com/mholt/certmagic.(*Config).getCertDuringHandshake(0xc0007e8000, 0xc0007e20b0, 0x101, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/damnever/.go/pkg/mod/github.com/mholt/certmagic@v0.9.3/handshake.go:206 +0xd2
github.com/mholt/certmagic.(*Config).GetCertificate(0xc0007e8000, 0xc0007e20b0, 0x0, 0x0, 0x0)
        /Users/damnever/.go/pkg/mod/github.com/mholt/certmagic@v0.9.3/handshake.go:72 +0x17d
github.com/caddyserver/caddy/v2/modules/caddytls.(*ConnectionPolicy).buildStandardTLSConfig.func1(0xc0007e20b0, 0xc000677a10, 0x1045128, 0xb0)
        /Users/damnever/.go/pkg/mod/github.com/caddyserver/caddy/v2@v2.0.0-beta.14/modules/caddytls/connpolicy.go:169 +0x36a
github.com/marten-seemann/qtls.(*Config).getCertificate(0xc0007e4000, 0xc0007e20b0, 0xc0007e20b0, 0xc00048a2e0, 0x5)
        /Users/damnever/.go/pkg/mod/github.com/marten-seemann/qtls@v0.4.1/common.go:896 +0x5ac
github.com/marten-seemann/qtls.(*serverHandshakeStateTLS13).pickCertificate(0xc0005e5d88, 0x0, 0x0)
        /Users/damnever/.go/pkg/mod/github.com/marten-seemann/qtls@v0.4.1/handshake_server_tls13.go:372 +0xf6
github.com/marten-seemann/qtls.(*serverHandshakeStateTLS13).handshake(0xc0005e5d88, 0xc0007e0000, 0x0)
        /Users/damnever/.go/pkg/mod/github.com/marten-seemann/qtls@v0.4.1/handshake_server_tls13.go:53 +0xb7
github.com/marten-seemann/qtls.(*Conn).serverHandshake(0xc0007a4000, 0x0, 0x18)
        /Users/damnever/.go/pkg/mod/github.com/marten-seemann/qtls@v0.4.1/handshake_server.go:54 +0x157
github.com/marten-seemann/qtls.(*Conn).Handshake(0xc0007a4000, 0x0, 0x0)
        /Users/damnever/.go/pkg/mod/github.com/marten-seemann/qtls@v0.4.1/conn.go:1415 +0x3cd
github.com/lucas-clemente/quic-go/internal/handshake.(*cryptoSetup).RunHandshake.func1(0xc00079a120, 0xc0000c6540, 0xc0002cb7a0)
        /Users/damnever/.go/pkg/mod/github.com/lucas-clemente/quic-go@v0.14.1/internal/handshake/crypto_setup.go:239 +0x9f
created by github.com/lucas-clemente/quic-go/internal/handshake.(*cryptoSetup).RunHandshake
        /Users/damnever/.go/pkg/mod/github.com/lucas-clemente/quic-go@v0.14.1/internal/handshake/crypto_setup.go:237 +0xbd
```

I have found that the [hello.Conn.LocalAddr()](https://github.com/mholt/certmagic/blob/v0.9.3/handshake.go#L101) is nil
```
&{CipherSuites:[49199 49200 49195 49196 52392 52393 49171 49161 49172 49162 156 157 47 53 49170 10 4865 4867 4866] ServerName: SupportedCurves:[29 23 24 25] SupportedPoints:[0] SignatureSchemes:[2052 1027 2055 2053 2054 1025 1281 15
37 1283 1539 513 515] SupportedProtos:[h3-24] SupportedVersions:[772] Conn:0xc0005c03f0}

Local:  <nil>
Remote:  127.0.0.1:51409
```

Hope the above information is useful.

--- a/go.mod
+++ b/go.mod
@@ -16,7 +16,7 @@ require (
 	github.com/klauspost/compress v1.8.6
 	github.com/klauspost/cpuid v1.2.2
 	github.com/kylelemons/godebug v1.1.0 // indirect
-	github.com/lucas-clemente/quic-go v0.14.1
+	github.com/lucas-clemente/quic-go v0.14.4
 	github.com/mholt/certmagic v0.9.3
 	github.com/miekg/dns v1.1.25 // indirect
 	github.com/muhammadmuzzammil1998/jsonc v0.0.0-20190906142622-1265e9b150c6

